activerecord/lib/active_record/railties/databases.rake

```diff
@@ -107,9 +110,12 @@ db_namespace = namespace :db do
     ActiveRecord::Tasks::DatabaseTasks.for_each(databases) do |name|
       # IMPORTANT: This task won't dump the schema if ActiveRecord::Base.dump_schema_after_migration is set to false
       task name do
-        if ActiveRecord::Base.dump_schema_after_migration
+        db_config = ActiveRecord::DatabaseConfigurations.new(databases).configs_for(env_name: Rails.env, name: name)
+
+        if db_config.schema_dump && ActiveRecord::Base.dump_schema_after_migration
```

Comment:
Is there a reason this conditional is different from the other task? Can you switch this one for consistency?